src/routes/safe/store/reducer/transactions.ts

```diff
@@ -1,13 +1,70 @@
 import { Map } from 'immutable'
 import { handleActions } from 'redux-actions'
 
-import { ADD_TRANSACTIONS } from 'src/routes/safe/store/actions/addTransactions'
+import { ADD_OR_UPDATE_TRANSACTIONS } from 'src/routes/safe/store/actions/transactions/addOrUpdateTransactions'
+import { REMOVE_TRANSACTION } from 'src/routes/safe/store/actions/transactions/removeTransaction'
 
 export const TRANSACTIONS_REDUCER_ID = 'transactions'
 
 export default handleActions(
   {
-    [ADD_TRANSACTIONS]: (state, action) => action.payload,
+    [ADD_OR_UPDATE_TRANSACTIONS]: (state, action) => {
```

Comment:
sorry if I've already asked this, why do we have merged action for merging/updating transactions?

I think the purpose of actions to be small so they're easy to debug, and debugging this particular one would be very hard

Reply:
I'm not sure if I agree. It was easy for me to debug, also I was following the recommendation of putting as much logic as possible inside a reducer.

As far as I understand, in the scenario you're purposing I need to know beforehand if the action I need to call is an update or addition. Requiring me to move and, probably, replicate the logic that is inside the reducer to all the places where the action is being called.

Reply:
> in the scenario you're purposing I need to know beforehand if the action I need to call is an update or addition.

Yeah, we should set a goal for us to make it possible this way